### PR TITLE
Add org.btchd.btchd-qt

### PR DESCRIPTION
**Pull request by benthetechguy**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

I don't personally own the domain but I know the people that do and they can make any necessary changes for me.

- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

The additional files are `org.btchd.btchd-qt.metainfo.xml`, `org.btchd.btchd-qt.desktop`, and `install_wrappers.sh`, and they have not been upstreamed because they would change the way that the normal application is built.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by benthetechguy** [comment]
bot, build org.btchd.btchd-qt

**Comment by benthetechguy** [comment]
bot, build org.btchd.btchd-qt

File: org.btchd.btchd-qt.json [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          "only-arches": ["i386"],
+          "url": "https://github.com/btchd/btchd/releases/download/v1.4.3/bhd-v1.4.3-7fe720529-i686-linux-gnu.tar.gz",
+          "sha256": "f9a435277a30ee075da989bf13fdfaa4b2c6fd20e274252c6936acf9721f281f"
+        },
+        {
+          "type": "archive",
+          "only-arches": ["arm"],
+          "url": "https://github.com/btchd/btchd/releases/download/v1.4.3/bhd-v1.4.3-7fe720529-arm-linux-gnueabihf.tar.gz",
+          "sha256": "1e5abee5de2533fb60ada173314fdd710692ac6ec41d0076b71585202295b331"
+        },
```
Comment (hfiguiere) [inline org.btchd.btchd-qt.json:49]: neither i386 nor arm are supported by the runtime, so this is pointless.

File: org.btchd.btchd-qt.json [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        {
+          "type": "archive",
+          "only-arches": ["aarch64"],
+          "url": "https://github.com/btchd/btchd/releases/download/v1.4.3/bhd-v1.4.3-7fe720529-aarch64-linux-gnu.tar.gz",
+          "sha256": "5640f2115e6e38e4124442b04d58d06f7abfed84a41fab81c1b2900618344bb3"
+        }
```
Comment (hfiguiere) [inline org.btchd.btchd-qt.json:55]: couldn't this be built from source instead?

File: org.btchd.btchd-qt.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    "--socket=pulseaudio",
+    /* OpenGL access. */
+    "--device=dri",
+    /* IPC access. */
+    "--share=ipc",
+    /* Network acess. */
+    "--share=network",
+    /* Tor auth cookie */
+    "--filesystem=/run/tor/control.authcookie:ro",
+    "--filesystem=/var/run/tor/control.authcookie:ro"
```
Comment (hfiguiere) [inline org.btchd.btchd-qt.json:20]: what does this do?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.btchd.btchd-qt.json (hunk 3 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.json:20]: No idea, got it from [the Bitcoin package](https://github.com/flathub/org.bitcoincore.bitcoin-qt/blob/master/org.bitcoincore.bitcoin-qt.json).

File: org.btchd.btchd-qt.json (hunk 2 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.json:55]: The source currently has issues building the GUI. The Bitcoin Package also uses binaries so I assumed it was okay to do the same for mine.

File: org.btchd.btchd-qt.json (hunk 3 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.json]: Removed it for now to be safe.

File: org.btchd.btchd-qt.json (hunk 1 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.json]: Removed.

**Comment by benthetechguy** [comment]
bot, build org.btchd.btchd-qt

File: org.btchd.btchd-qt.json (hunk 3 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.json]: Update: It's needed to allow some things to be accessed outside the sandbox. [From the wiki](https://docs.flatpak.org/en/latest/manifests.html#finishing). I'll research it more and add it back with the right options.

Comment (hfiguiere) [reply org.btchd.btchd-qt.json:20]: what things?

Comment (benthetechguy) [reply org.btchd.btchd-qt.json:20]: X11 support so it can have a GUI, PulseAudio for sound, OpenGL, and the most important being network access.

Comment (hfiguiere) [reply org.btchd.btchd-qt.json:20]: talking about line 19 and 20.

Comment (benthetechguy) [reply org.btchd.btchd-qt.json:20]: https://github.com/btchd/btchd/blob/master/doc/tor.md

**Comment by hfiguiere** [comment]
bot, build org.btchd.btchd-qt

File: org.btchd.btchd-qt.json (hunk 3 above)
Comment (barthalion) [inline org.btchd.btchd-qt.json:20]:
```suggestion
```

I think all modern distros make `/var/run` a symlink to `/run`, and it's not going to launch if that's the case.

File: org.btchd.btchd-qt.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  <project_license>MIT</project_license>
+  <metadata_license>CC0-1.0</metadata_license>
+  <summary>
+    Connect to the BitcoinHD P2P network
+  </summary>
+  <url type="homepage">
+    https://btchd.org/
+  </url>
+  <url type="bugtracker">
+    https://github.com/bitcoinhd/bitcoinhd/issues/new?title=[flatpak]
```
Comment (barthalion) [inline org.btchd.btchd-qt.metainfo.xml:14]:
```suggestion
    https://github.com/btchd/btchd/issues/new?title=[flatpak]
```

File: org.btchd.btchd-qt.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  </launchable>
+  <description>
+    <p>
+    Bitcoin is an experimental digital currency that enables instant
+    payments to anyone, anywhere in the world. Bitcoin uses
+    peer-to-peer technology to operate with no central authority:
+    managing transactions and issuing money are carried out
+    collectively by the network. Bitcoin Core is the name of open
+    source software which enables the use of this currency.
+    </p>
```
Comment (barthalion) [inline org.btchd.btchd-qt.metainfo.xml:27]: I would add a section here explaining how BTCHD is different from regular Bitcoin.

File: org.btchd.btchd-qt.desktop [hunk 6]
```diff
@@ -0,0 +1,14 @@
+[Desktop Entry]
```
Comment (barthalion) [inline org.btchd.btchd-qt.desktop:1]: FYI ideally you should maintain it upstream.

File: org.btchd.btchd-qt.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline org.btchd.btchd-qt.metainfo.xml:1]: FYI ideally you should maintain it upstream.

File: org.btchd.btchd-qt.metainfo.xml (hunk 4 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.metainfo.xml:14]: Oops, didn't catch that one

File: org.btchd.btchd-qt.desktop (hunk 6 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.desktop:1]: I asked but the owner is opposed to it for whatever reason.

File: org.btchd.btchd-qt.metainfo.xml (hunk 7 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.metainfo.xml:1]: I asked but the owner is opposed to it for whatever reason.

**Comment by benthetechguy** [comment]
bot, build org.btchd.btchd-qt

File: org.btchd.btchd-qt.metainfo.xml (hunk 5 above)
Comment (benthetechguy) [reply org.btchd.btchd-qt.metainfo.xml]: Done.

**Comment by benthetechguy** [comment]
bot, build org.btchd.btchd-qt

**Pull request closed without merging**